core/src/main/java/org/mule/runtime/core/internal/event/AbstractEventContext.java

```diff
@@ -153,40 +129,101 @@ public final void success(CoreEvent event) {
         return just((MessagingException) throwable)
             .flatMapMany(exceptionHandler)
             .doOnNext(handled -> success(handled))
-            .doOnError(rethrown -> responseProcessor.onError(rethrown))
+            .doOnError(rethrown -> responseDone(null, rethrown))
             .materialize()
             .then()
             .toProcessor();
       } else {
-        responseProcessor.onError(throwable);
+        responseDone(null, throwable);
         return empty();
       }
     }
   }
 
+  private synchronized void responseDone(CoreEvent event, Throwable throwable) {
+    this.state = STATE_DONE;
+    onResponseConsumerList.stream().forEach(consumer -> consumer.accept(event, throwable));
```

Comment:
what if one of these fail? Shouldn't the rest of this method be executed anyway? Or suppose you have 3 consumers, if one fails, shouldn't the rest be invoked also?